sdks/wasm/bindings.cs

```diff
@@ -0,0 +1,174 @@
+using System;
+using System.Text;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Reflection;
+using NUnitLite.Runner;
+using NUnit.Framework.Internal;
+using NUnit.Framework.Api;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+
+namespace WebAssembly {
+	/*
+	TODO:
+		Expose annotated C# type to JS
+		Add property fetch to JSObject
+		Add typed method invoke support (get a delegate?)
+		Add JS helpers to fetch wrapped methods, like to Module.cwrap
+		Better Wrap C# exception when passing them as object (IE, on task failure)
+		Make JSObject disposable (same for js objects)
+	*/
+	public sealed class Runtime {
+		[MethodImplAttribute (MethodImplOptions.InternalCall)]
+		static extern string InvokeJS (string str, out int exceptional_result);
+
+		[MethodImplAttribute (MethodImplOptions.InternalCall)]
+		internal static extern object InvokeJSWithArgs (int js_obj_handle, string method, object[] _params, out int exceptional_result);
```

Comment:
Are these all safe for concurrent access? I can't quite picture where you'd need to lock, but I feel like I see some accesses and mutations that would be unsafe concurrently. Things like `(int)(IntPtr)obj.Handle`.

Reply:
So, JS is single threaded so this is not an issue today.
Even later, with the multithreaded wasm proposal, the JS heap will only be accessible from the main thread.